lightningd/peer_control.c

```diff
@@ -1107,6 +1107,47 @@ peer_connected_hook_deserialize(struct peer_connected_hook_payload *payload,
 	return true;
 }
 
+/* Compare and store `remote_addr` and the `peer_id` that reported it.
+ * If new address was reported by at least one other, do node_announcement */
+static void update_remote_addr(struct lightningd *ld,
+			       struct wireaddr *remote_addr,
+			       struct node_id peer_id)
```

Comment:
*const* struct wireaddr.  And pass node_id by copy, so "const struct node_id *".